sway/desktop/fx_renderer.c

```diff
@@ -449,10 +451,16 @@ bool fx_render_subtexture_with_matrix(struct fx_renderer *renderer, struct wlr_t
 
 	float* dim_color = deco_data.dim_color;
 
+	struct wlr_box transformed_box;
+	int width, height;
+	wlr_output_transformed_resolution(wlr_output, &width, &height);
+	wlr_box_transform(&transformed_box, dst_box,
+			wlr_output_transform_invert(wlr_output->transform), width, height);
+
 	glUniformMatrix3fv(shader->proj, 1, GL_FALSE, gl_matrix);
 	glUniform1i(shader->tex, 0);
-	glUniform2f(shader->size, dst_box->width, dst_box->height);
-	glUniform2f(shader->position, dst_box->x, dst_box->y);
+	glUniform2f(shader->size, transformed_box.width, transformed_box.height);
+	glUniform2f(shader->position, transformed_box.x, transformed_box.y);
```

Comment:
Let's instead transform the box in render.c, and pass the transformed box as dst_box. Same goes for the other fx rendering functions.